From 00c315cb0c8ab77c67be4a2ac08a71a83ac58751 Mon Sep 17 00:00:00 2001
From: Hans Zandbelt <hans.zandbelt@zmartzone.eu>
Date: Fri, 25 Jun 2021 11:42:57 +0200
Subject: [PATCH] avoid XSS vulnerability when using OIDCPreservePost On

and supplying URLs that contain single quotes; thanks @oss-aimoto

Signed-off-by: Hans Zandbelt <hans.zandbelt@zmartzone.eu>
---
 ChangeLog              | 4 ++++
 src/mod_auth_openidc.c | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/ChangeLog b/ChangeLog
index 038e47d3..27d5b865 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,7 @@
+06/25/2021
+- avoid XSS vulnerability when using OIDCPreservePost On and supplying URLs that contain single quotes
+  thanks @oss-aimoto
+
 06/21/2021
 - strip A256GCM JWT header from encrypted JWTS used for state cookies, cache encryption and by-value session cookies
   resulting in smaller cookies and reduced cache content size
diff --git a/src/mod_auth_openidc.c b/src/mod_auth_openidc.c
index b1f99a6b..4f94743b 100644
--- a/src/mod_auth_openidc.c
+++ b/src/mod_auth_openidc.c
@@ -513,7 +513,7 @@ static int oidc_request_post_preserved_restore(request_rec *r,
 					"          input.type = \"hidden\";\n"
 					"          document.forms[0].appendChild(input);\n"
 					"        }\n"
-					"        document.forms[0].action = '%s';\n"
+					"        document.forms[0].action = \"%s\";\n"
 					"        document.forms[0].submit();\n"
 					"      }\n"
 					"    </script>\n", method, original_url);
